src/content/en/updates/2017/09/abortable-fetch.md

```diff
@@ -0,0 +1,305 @@
+couple of false starts, and 5 pull requests, we finally have abortable fetch landing in browsers,
+with the first being Firefox 57.
+
+I'll dive into the history later, but first, the API:
+
+## The controller + signal manoeuvre
+
+Meet the `AbortController` and `AbortSignal`:
+
+    const controller = new AbortController();
+    const signal = controller.signal;
+
+The controller only has one method:
+
+    controller.abort();
+
+When you do this, it notifies the signal:
+
+    signal.addEventListener('abort', () => {
+      // Logs true:
+      console.log(signal.aborted);
+    });
+
+This API is provided by [the DOM
+standard](https://dom.spec.whatwg.org/#aborting-ongoing-activities), and that's the entire API. It's
+deliberately generic so it can be used by other web standards and JavaScript libraries.
+
+## Abort signals and fetch
+
+Fetch can take an `AbortSignal`. For instance, here's how you'd make a fetch timeout after 5
+seconds:
+
+    const controller = new AbortController();
+    const signal = controller.signal;
+
+    setTimeout(() => controller.abort(), 5000);
+
+    fetch(url, { signal }).then(response => {
+      return response.text();
+    }).then(text => {
+      console.log(text);
+    });
+
+When you abort a fetch, it aborts both the request and response, so any reading of the response body
+(such as `response.text()`) is also aborted.
+
+Note: It's ok to call `.abort()` after the fetch has already completed, fetch simply ignores it.
+
+**[Here's a demo](https://fetch-abort-demo.glitch.me/)** – At time of writing, the only browser
+which supports this is Firefox 57. Also, brace yourself, no one with any design skill was involved
+in creating the demo.
+
+Alternatively, the signal can be given to a request object and later passed to fetch:
+
+    const controller = new AbortController();
+    const signal = controller.signal;
+    const request = new Request(url, { signal });
+
+    fetch(request);
+
+This works because `request.signal` is an `AbortSignal`.
+
+Note: Technically, `request.signal` isn't the same signal you pass to the constructor. It's a new
+`AbortSignal` that mimics the signal passed to the constructor. This means every `Request` has a
+signal, whether one is given to its constructor or not.
+
+## Reacting to an aborted fetch
+
+When you abort an async operation, the promise rejects with a `DOMException` named `AbortError`:
+
+    fetch(url, { signal }).then(response => {
+      return response.text();
+    }).then(text => {
+      console.log(text);
+    }).catch(err => {
+      if (err.name === 'AbortError') {
+        console.log('Fetch aborted');
+      } else {
+        console.log('Uh oh, an error!', err);
+      }
+    });
+
+You don't often want to show an error message if the user aborted the operation, as it isn't an
+"error" if you successfully do that the user asked. To avoid this, use an if-statement such as the
+one above to handle abort errors specifically.
```

Comment:
This reminds me of a bug report we had.  When a user navigates on their site they would get a NetworkError rejection for their fetch() which would cause them to flash an error before the new window opened.  Perhaps now we can reject with an AbortError so they can ignore it like you suggest here.

Reply:
Yeah, we should define what UA fetch terminations look like.